### PR TITLE
Update gradlew to 6.2.2

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
Update to the latest version of gradlew and update dependencies.

Modifications:
- Use spotbugs 4.0.2 plugin and 4.0.0 executable
- Remove explicit guava compile dependency

Result:
Project now uses gradlew 6.2.2.

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -31,10 +31,6 @@ apply plugin: "java-gradle-plugin"
 apply from: "../servicetalk-grpc-gradle-plugin/plugin-config.gradle"
 apply from: "../servicetalk-gradle-plugin-internal/plugin-config.gradle"
 
-dependencies {
-  compile "com.google.guava:guava:28.1-jre" // needed because of conflicts between Spotbugs and Protobuf
```
Comment (idelpivnitskiy) [inline buildSrc/build.gradle]: ❤️

**Comment by volyx** [comment]
I'm confused with "Integration with Intellij IDEA ✅". Is it possible to run test with Intellij IDEA? 

Now I'm running test with 
```
CI=true ./gradlew :servicetalk-http-router-predicate:test --tests HttpPredicateRouterBuilderTest
```

I have tried two ways of importing project to IDEA:
- with servicetalk.ipr(Skip gradle import)
- Import as gradle project

With two this approaches I can't run test through IDEA UI. I'm doing something wrong, aren't I? Could you help me please to find proper way to configure IDEA project

After `open servicetalk.ipr` IDEA asks about gradle. Should I import project as ipr or gradle?

image

<details><summary>IntelliJ IDEA 2019.3.3 (Community Edition)</summary>

IntelliJ IDEA 2019.3.3 (Community Edition)
Build #IC-193.6494.35, built on February 11, 2020
Runtime version: 11.0.5+10-b520.38 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.5
GC: ParNew, ConcurrentMarkSweep
Memory: 990M
Cores: 4
Registry: 
Non-Bundled Plugins: CheckStyle-IDEA, Dart, FindBugs-IDEA, com.azoft.json2dart, com.karateca.jstoolbox, com.zxk.plugin.buildjson, ideanginx9, io.flutter, io.protostuff.protostuff-jetbrains-plugin, mobi.hsz.idea.gitignore, name.kropp.intellij.makefile, ru.adelf.idea.dotenv

</details>

<details><summary>Java version</summary>
java -version
java version "1.8.0_141"
Java(TM) SE Runtime Environment (build 1.8.0_141-b15)
Java HotSpot(TM) 64-Bit Server VM (build 25.141-b15, mixed mode)
</details>

**Comment by idelpivnitskiy** [comment]
@volyx What I did during the latest testing:
1. `./gradlew clean cleanIdea cleanIdeaWorkspace cleanIdeaProject`
2. `./gradlew idea`
3. `open servicetalk.ipr`

Then IDE asks to import Gradle Project (as you show on the picture). Personally, I prefer to skip the gradle import for servicetalk because I saw different random glitches when you import large multi-module projects (especially with multiple source sets, like `main/test/testFixtures`). If I skip the import, every time I run a test in Intellij IDEA, the IDE runs it. I do not run all tests via Intellij IDEA. Usually run only a single test, or a single test class, or a single test package. All debugging features and break-points work in this mode. If I need to run all tests, I use command line: `./gradlew build`. All of these are just personal preferences and habits.

For this test I also tried to import the gradle project. Just to verify that it works. I tested that gradle /ide can resolve all dependencies, update them if I change `build.gradle` on the fly (if you do not import gradle project, you have to run `./gradlew idea` manually every time you change `build.gradle` files to see new depdencies in Intellij IDEA), and run tests. I saw that after gradle import intellij idea runs tests via gradlew (even individual tests), not via standard junit test configuration.

Tested with Intellij IDEA 2019.3.3 (Community Edition). Build #IC-193.6494.35.

**Comment by volyx** [comment]
@idelpivnitskiy  Steps `./gradlew clean cleanIdea cleanIdeaWorkspace cleanIdeaProject` and `skip the gradle ` help! Thanks!

**Comment by idelpivnitskiy** [comment]
I will leave an additional comment in this thread for records:

Importing the ST project as a Gradle project in Intellij IDEA is also fine. To configure the IDE to run tests instead of invoking Gradle go to Preferences -> Build, Execution, Deployment -> Build Tools -> Gradle. Choose the project and then configure “Build and run using” & “Run tests using” to use Intellij IDEA.

**Pull request merged**